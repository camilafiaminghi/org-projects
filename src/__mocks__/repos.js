export default {
	"total_count": 154,
	"incomplete_results": false,
	"items": [
		{
			"id": 10270250,
			"node_id": "MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==",
			"name": "react",
			"full_name": "facebook/react",
			"private": false,
			"owner": {
				"login": "facebook",
				"id": 69631,
				"node_id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
				"avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4",
				"gravatar_id": "",
				"url": "https://api.github.com/users/facebook",
				"html_url": "https://github.com/facebook",
				"followers_url": "https://api.github.com/users/facebook/followers",
				"following_url": "https://api.github.com/users/facebook/following{/other_user}",
				"gists_url": "https://api.github.com/users/facebook/gists{/gist_id}",
				"starred_url": "https://api.github.com/users/facebook/starred{/owner}{/repo}",
				"subscriptions_url": "https://api.github.com/users/facebook/subscriptions",
				"organizations_url": "https://api.github.com/users/facebook/orgs",
				"repos_url": "https://api.github.com/users/facebook/repos",
				"events_url": "https://api.github.com/users/facebook/events{/privacy}",
				"received_events_url": "https://api.github.com/users/facebook/received_events",
				"type": "Organization",
				"site_admin": false
			},
			"html_url": "https://github.com/facebook/react",
			"description": "A declarative, efficient, and flexible JavaScript library for building user interfaces.",
			"fork": false,
			"url": "https://api.github.com/repos/facebook/react",
			"forks_url": "https://api.github.com/repos/facebook/react/forks",
			"keys_url": "https://api.github.com/repos/facebook/react/keys{/key_id}",
			"collaborators_url": "https://api.github.com/repos/facebook/react/collaborators{/collaborator}",
			"teams_url": "https://api.github.com/repos/facebook/react/teams",
			"hooks_url": "https://api.github.com/repos/facebook/react/hooks",
			"issue_events_url": "https://api.github.com/repos/facebook/react/issues/events{/number}",
			"events_url": "https://api.github.com/repos/facebook/react/events",
			"assignees_url": "https://api.github.com/repos/facebook/react/assignees{/user}",
			"branches_url": "https://api.github.com/repos/facebook/react/branches{/branch}",
			"tags_url": "https://api.github.com/repos/facebook/react/tags",
			"blobs_url": "https://api.github.com/repos/facebook/react/git/blobs{/sha}",
			"git_tags_url": "https://api.github.com/repos/facebook/react/git/tags{/sha}",
			"git_refs_url": "https://api.github.com/repos/facebook/react/git/refs{/sha}",
			"trees_url": "https://api.github.com/repos/facebook/react/git/trees{/sha}",
			"statuses_url": "https://api.github.com/repos/facebook/react/statuses/{sha}",
			"languages_url": "https://api.github.com/repos/facebook/react/languages",
			"stargazers_url": "https://api.github.com/repos/facebook/react/stargazers",
			"contributors_url": "https://api.github.com/repos/facebook/react/contributors",
			"subscribers_url": "https://api.github.com/repos/facebook/react/subscribers",
			"subscription_url": "https://api.github.com/repos/facebook/react/subscription",
			"commits_url": "https://api.github.com/repos/facebook/react/commits{/sha}",
			"git_commits_url": "https://api.github.com/repos/facebook/react/git/commits{/sha}",
			"comments_url": "https://api.github.com/repos/facebook/react/comments{/number}",
			"issue_comment_url": "https://api.github.com/repos/facebook/react/issues/comments{/number}",
			"contents_url": "https://api.github.com/repos/facebook/react/contents/{+path}",
			"compare_url": "https://api.github.com/repos/facebook/react/compare/{base}...{head}",
			"merges_url": "https://api.github.com/repos/facebook/react/merges",
			"archive_url": "https://api.github.com/repos/facebook/react/{archive_format}{/ref}",
			"downloads_url": "https://api.github.com/repos/facebook/react/downloads",
			"issues_url": "https://api.github.com/repos/facebook/react/issues{/number}",
			"pulls_url": "https://api.github.com/repos/facebook/react/pulls{/number}",
			"milestones_url": "https://api.github.com/repos/facebook/react/milestones{/number}",
			"notifications_url": "https://api.github.com/repos/facebook/react/notifications{?since,all,participating}",
			"labels_url": "https://api.github.com/repos/facebook/react/labels{/name}",
			"releases_url": "https://api.github.com/repos/facebook/react/releases{/id}",
			"deployments_url": "https://api.github.com/repos/facebook/react/deployments",
			"created_at": "2013-05-24T16:15:54Z",
			"updated_at": "2019-02-09T23:36:29Z",
			"pushed_at": "2019-02-09T19:42:31Z",
			"git_url": "git://github.com/facebook/react.git",
			"ssh_url": "git@github.com:facebook/react.git",
			"clone_url": "https://github.com/facebook/react.git",
			"svn_url": "https://github.com/facebook/react",
			"homepage": "https://reactjs.org",
			"size": 140296,
			"stargazers_count": 121943,
			"watchers_count": 121943,
			"language": "JavaScript",
			"has_issues": true,
			"has_projects": true,
			"has_downloads": true,
			"has_wiki": true,
			"has_pages": true,
			"forks_count": 22100,
			"mirror_url": null,
			"archived": false,
			"open_issues_count": 538,
			"license": {
				"key": "mit",
				"name": "MIT License",
				"spdx_id": "MIT",
				"url": "https://api.github.com/licenses/mit",
				"node_id": "MDc6TGljZW5zZTEz"
			},
			"forks": 22100,
			"open_issues": 538,
			"watchers": 121943,
			"default_branch": "master",
			"score": 1.0
		},
		{
			"id": 29028775,
			"node_id": "MDEwOlJlcG9zaXRvcnkyOTAyODc3NQ==",
			"name": "react-native",
			"full_name": "facebook/react-native",
			"private": false,
			"owner": {
				"login": "facebook",
				"id": 69631,
				"node_id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
				"avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4",
				"gravatar_id": "",
				"url": "https://api.github.com/users/facebook",
				"html_url": "https://github.com/facebook",
				"followers_url": "https://api.github.com/users/facebook/followers",
				"following_url": "https://api.github.com/users/facebook/following{/other_user}",
				"gists_url": "https://api.github.com/users/facebook/gists{/gist_id}",
				"starred_url": "https://api.github.com/users/facebook/starred{/owner}{/repo}",
				"subscriptions_url": "https://api.github.com/users/facebook/subscriptions",
				"organizations_url": "https://api.github.com/users/facebook/orgs",
				"repos_url": "https://api.github.com/users/facebook/repos",
				"events_url": "https://api.github.com/users/facebook/events{/privacy}",
				"received_events_url": "https://api.github.com/users/facebook/received_events",
				"type": "Organization",
				"site_admin": false
			},
			"html_url": "https://github.com/facebook/react-native",
			"description": "A framework for building native apps with React.",
			"fork": false,
			"url": "https://api.github.com/repos/facebook/react-native",
			"forks_url": "https://api.github.com/repos/facebook/react-native/forks",
			"keys_url": "https://api.github.com/repos/facebook/react-native/keys{/key_id}",
			"collaborators_url": "https://api.github.com/repos/facebook/react-native/collaborators{/collaborator}",
			"teams_url": "https://api.github.com/repos/facebook/react-native/teams",
			"hooks_url": "https://api.github.com/repos/facebook/react-native/hooks",
			"issue_events_url": "https://api.github.com/repos/facebook/react-native/issues/events{/number}",
			"events_url": "https://api.github.com/repos/facebook/react-native/events",
			"assignees_url": "https://api.github.com/repos/facebook/react-native/assignees{/user}",
			"branches_url": "https://api.github.com/repos/facebook/react-native/branches{/branch}",
			"tags_url": "https://api.github.com/repos/facebook/react-native/tags",
			"blobs_url": "https://api.github.com/repos/facebook/react-native/git/blobs{/sha}",
			"git_tags_url": "https://api.github.com/repos/facebook/react-native/git/tags{/sha}",
			"git_refs_url": "https://api.github.com/repos/facebook/react-native/git/refs{/sha}",
			"trees_url": "https://api.github.com/repos/facebook/react-native/git/trees{/sha}",
			"statuses_url": "https://api.github.com/repos/facebook/react-native/statuses/{sha}",
			"languages_url": "https://api.github.com/repos/facebook/react-native/languages",
			"stargazers_url": "https://api.github.com/repos/facebook/react-native/stargazers",
			"contributors_url": "https://api.github.com/repos/facebook/react-native/contributors",
			"subscribers_url": "https://api.github.com/repos/facebook/react-native/subscribers",
			"subscription_url": "https://api.github.com/repos/facebook/react-native/subscription",
			"commits_url": "https://api.github.com/repos/facebook/react-native/commits{/sha}",
			"git_commits_url": "https://api.github.com/repos/facebook/react-native/git/commits{/sha}",
			"comments_url": "https://api.github.com/repos/facebook/react-native/comments{/number}",
			"issue_comment_url": "https://api.github.com/repos/facebook/react-native/issues/comments{/number}",
			"contents_url": "https://api.github.com/repos/facebook/react-native/contents/{+path}",
			"compare_url": "https://api.github.com/repos/facebook/react-native/compare/{base}...{head}",
			"merges_url": "https://api.github.com/repos/facebook/react-native/merges",
			"archive_url": "https://api.github.com/repos/facebook/react-native/{archive_format}{/ref}",
			"downloads_url": "https://api.github.com/repos/facebook/react-native/downloads",
			"issues_url": "https://api.github.com/repos/facebook/react-native/issues{/number}",
			"pulls_url": "https://api.github.com/repos/facebook/react-native/pulls{/number}",
			"milestones_url": "https://api.github.com/repos/facebook/react-native/milestones{/number}",
			"notifications_url": "https://api.github.com/repos/facebook/react-native/notifications{?since,all,participating}",
			"labels_url": "https://api.github.com/repos/facebook/react-native/labels{/name}",
			"releases_url": "https://api.github.com/repos/facebook/react-native/releases{/id}",
			"deployments_url": "https://api.github.com/repos/facebook/react-native/deployments",
			"created_at": "2015-01-09T18:10:16Z",
			"updated_at": "2019-02-10T00:00:59Z",
			"pushed_at": "2019-02-09T18:18:11Z",
			"git_url": "git://github.com/facebook/react-native.git",
			"ssh_url": "git@github.com:facebook/react-native.git",
			"clone_url": "https://github.com/facebook/react-native.git",
			"svn_url": "https://github.com/facebook/react-native",
			"homepage": "https://facebook.github.io/react-native/",
			"size": 311684,
			"stargazers_count": 73686,
			"watchers_count": 73686,
			"language": "JavaScript",
			"has_issues": true,
			"has_projects": true,
			"has_downloads": true,
			"has_wiki": true,
			"has_pages": true,
			"forks_count": 16433,
			"mirror_url": null,
			"archived": false,
			"open_issues_count": 764,
			"license": {
				"key": "other",
				"name": "Other",
				"spdx_id": "NOASSERTION",
				"url": null,
				"node_id": "MDc6TGljZW5zZTA="
			},
			"forks": 16433,
			"open_issues": 764,
			"watchers": 73686,
			"default_branch": "master",
			"score": 1.0
		},
		{
			"id": 63537249,
			"node_id": "MDEwOlJlcG9zaXRvcnk2MzUzNzI0OQ==",
			"name": "create-react-app",
			"full_name": "facebook/create-react-app",
			"private": false,
			"owner": {
				"login": "facebook",
				"id": 69631,
				"node_id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
				"avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4",
				"gravatar_id": "",
				"url": "https://api.github.com/users/facebook",
				"html_url": "https://github.com/facebook",
				"followers_url": "https://api.github.com/users/facebook/followers",
				"following_url": "https://api.github.com/users/facebook/following{/other_user}",
				"gists_url": "https://api.github.com/users/facebook/gists{/gist_id}",
				"starred_url": "https://api.github.com/users/facebook/starred{/owner}{/repo}",
				"subscriptions_url": "https://api.github.com/users/facebook/subscriptions",
				"organizations_url": "https://api.github.com/users/facebook/orgs",
				"repos_url": "https://api.github.com/users/facebook/repos",
				"events_url": "https://api.github.com/users/facebook/events{/privacy}",
				"received_events_url": "https://api.github.com/users/facebook/received_events",
				"type": "Organization",
				"site_admin": false
			},
			"html_url": "https://github.com/facebook/create-react-app",
			"description": "Set up a modern web app by running one command.",
			"fork": false,
			"url": "https://api.github.com/repos/facebook/create-react-app",
			"forks_url": "https://api.github.com/repos/facebook/create-react-app/forks",
			"keys_url": "https://api.github.com/repos/facebook/create-react-app/keys{/key_id}",
			"collaborators_url": "https://api.github.com/repos/facebook/create-react-app/collaborators{/collaborator}",
			"teams_url": "https://api.github.com/repos/facebook/create-react-app/teams",
			"hooks_url": "https://api.github.com/repos/facebook/create-react-app/hooks",
			"issue_events_url": "https://api.github.com/repos/facebook/create-react-app/issues/events{/number}",
			"events_url": "https://api.github.com/repos/facebook/create-react-app/events",
			"assignees_url": "https://api.github.com/repos/facebook/create-react-app/assignees{/user}",
			"branches_url": "https://api.github.com/repos/facebook/create-react-app/branches{/branch}",
			"tags_url": "https://api.github.com/repos/facebook/create-react-app/tags",
			"blobs_url": "https://api.github.com/repos/facebook/create-react-app/git/blobs{/sha}",
			"git_tags_url": "https://api.github.com/repos/facebook/create-react-app/git/tags{/sha}",
			"git_refs_url": "https://api.github.com/repos/facebook/create-react-app/git/refs{/sha}",
			"trees_url": "https://api.github.com/repos/facebook/create-react-app/git/trees{/sha}",
			"statuses_url": "https://api.github.com/repos/facebook/create-react-app/statuses/{sha}",
			"languages_url": "https://api.github.com/repos/facebook/create-react-app/languages",
			"stargazers_url": "https://api.github.com/repos/facebook/create-react-app/stargazers",
			"contributors_url": "https://api.github.com/repos/facebook/create-react-app/contributors",
			"subscribers_url": "https://api.github.com/repos/facebook/create-react-app/subscribers",
			"subscription_url": "https://api.github.com/repos/facebook/create-react-app/subscription",
			"commits_url": "https://api.github.com/repos/facebook/create-react-app/commits{/sha}",
			"git_commits_url": "https://api.github.com/repos/facebook/create-react-app/git/commits{/sha}",
			"comments_url": "https://api.github.com/repos/facebook/create-react-app/comments{/number}",
			"issue_comment_url": "https://api.github.com/repos/facebook/create-react-app/issues/comments{/number}",
			"contents_url": "https://api.github.com/repos/facebook/create-react-app/contents/{+path}",
			"compare_url": "https://api.github.com/repos/facebook/create-react-app/compare/{base}...{head}",
			"merges_url": "https://api.github.com/repos/facebook/create-react-app/merges",
			"archive_url": "https://api.github.com/repos/facebook/create-react-app/{archive_format}{/ref}",
			"downloads_url": "https://api.github.com/repos/facebook/create-react-app/downloads",
			"issues_url": "https://api.github.com/repos/facebook/create-react-app/issues{/number}",
			"pulls_url": "https://api.github.com/repos/facebook/create-react-app/pulls{/number}",
			"milestones_url": "https://api.github.com/repos/facebook/create-react-app/milestones{/number}",
			"notifications_url": "https://api.github.com/repos/facebook/create-react-app/notifications{?since,all,participating}",
			"labels_url": "https://api.github.com/repos/facebook/create-react-app/labels{/name}",
			"releases_url": "https://api.github.com/repos/facebook/create-react-app/releases{/id}",
			"deployments_url": "https://api.github.com/repos/facebook/create-react-app/deployments",
			"created_at": "2016-07-17T14:55:11Z",
			"updated_at": "2019-02-10T00:01:02Z",
			"pushed_at": "2019-02-09T22:39:07Z",
			"git_url": "git://github.com/facebook/create-react-app.git",
			"ssh_url": "git@github.com:facebook/create-react-app.git",
			"clone_url": "https://github.com/facebook/create-react-app.git",
			"svn_url": "https://github.com/facebook/create-react-app",
			"homepage": "https://facebook.github.io/create-react-app/",
			"size": 8985,
			"stargazers_count": 63254,
			"watchers_count": 63254,
			"language": "JavaScript",
			"has_issues": true,
			"has_projects": true,
			"has_downloads": true,
			"has_wiki": false,
			"has_pages": true,
			"forks_count": 13896,
			"mirror_url": null,
			"archived": false,
			"open_issues_count": 356,
			"license": {
				"key": "mit",
				"name": "MIT License",
				"spdx_id": "MIT",
				"url": "https://api.github.com/licenses/mit",
				"node_id": "MDc6TGljZW5zZTEz"
			},
			"forks": 13896,
			"open_issues": 356,
			"watchers": 63254,
			"default_branch": "master",
			"score": 1.0
		},
		{
			"id": 21413198,
			"node_id": "MDEwOlJlcG9zaXRvcnkyMTQxMzE5OA==",
			"name": "immutable-js",
			"full_name": "facebook/immutable-js",
			"private": false,
			"owner": {
				"login": "facebook",
				"id": 69631,
				"node_id": "MDEyOk9yZ2FuaXphdGlvbjY5NjMx",
				"avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4",
				"gravatar_id": "",
				"url": "https://api.github.com/users/facebook",
				"html_url": "https://github.com/facebook",
				"followers_url": "https://api.github.com/users/facebook/followers",
				"following_url": "https://api.github.com/users/facebook/following{/other_user}",
				"gists_url": "https://api.github.com/users/facebook/gists{/gist_id}",
				"starred_url": "https://api.github.com/users/facebook/starred{/owner}{/repo}",
				"subscriptions_url": "https://api.github.com/users/facebook/subscriptions",
				"organizations_url": "https://api.github.com/users/facebook/orgs",
				"repos_url": "https://api.github.com/users/facebook/repos",
				"events_url": "https://api.github.com/users/facebook/events{/privacy}",
				"received_events_url": "https://api.github.com/users/facebook/received_events",
				"type": "Organization",
				"site_admin": false
			},
			"html_url": "https://github.com/facebook/immutable-js",
			"description": "Immutable persistent data collections for Javascript which increase efficiency and simplicity.",
			"fork": false,
			"url": "https://api.github.com/repos/facebook/immutable-js",
			"forks_url": "https://api.github.com/repos/facebook/immutable-js/forks",
			"keys_url": "https://api.github.com/repos/facebook/immutable-js/keys{/key_id}",
			"collaborators_url": "https://api.github.com/repos/facebook/immutable-js/collaborators{/collaborator}",
			"teams_url": "https://api.github.com/repos/facebook/immutable-js/teams",
			"hooks_url": "https://api.github.com/repos/facebook/immutable-js/hooks",
			"issue_events_url": "https://api.github.com/repos/facebook/immutable-js/issues/events{/number}",
			"events_url": "https://api.github.com/repos/facebook/immutable-js/events",
			"assignees_url": "https://api.github.com/repos/facebook/immutable-js/assignees{/user}",
			"branches_url": "https://api.github.com/repos/facebook/immutable-js/branches{/branch}",
			"tags_url": "https://api.github.com/repos/facebook/immutable-js/tags",
			"blobs_url": "https://api.github.com/repos/facebook/immutable-js/git/blobs{/sha}",
			"git_tags_url": "https://api.github.com/repos/facebook/immutable-js/git/tags{/sha}",
			"git_refs_url": "https://api.github.com/repos/facebook/immutable-js/git/refs{/sha}",
			"trees_url": "https://api.github.com/repos/facebook/immutable-js/git/trees{/sha}",
			"statuses_url": "https://api.github.com/repos/facebook/immutable-js/statuses/{sha}",
			"languages_url": "https://api.github.com/repos/facebook/immutable-js/languages",
			"stargazers_url": "https://api.github.com/repos/facebook/immutable-js/stargazers",
			"contributors_url": "https://api.github.com/repos/facebook/immutable-js/contributors",
			"subscribers_url": "https://api.github.com/repos/facebook/immutable-js/subscribers",
			"subscription_url": "https://api.github.com/repos/facebook/immutable-js/subscription",
			"commits_url": "https://api.github.com/repos/facebook/immutable-js/commits{/sha}",
			"git_commits_url": "https://api.github.com/repos/facebook/immutable-js/git/commits{/sha}",
			"comments_url": "https://api.github.com/repos/facebook/immutable-js/comments{/number}",
			"issue_comment_url": "https://api.github.com/repos/facebook/immutable-js/issues/comments{/number}",
			"contents_url": "https://api.github.com/repos/facebook/immutable-js/contents/{+path}",
			"compare_url": "https://api.github.com/repos/facebook/immutable-js/compare/{base}...{head}",
			"merges_url": "https://api.github.com/repos/facebook/immutable-js/merges",
			"archive_url": "https://api.github.com/repos/facebook/immutable-js/{archive_format}{/ref}",
			"downloads_url": "https://api.github.com/repos/facebook/immutable-js/downloads",
			"issues_url": "https://api.github.com/repos/facebook/immutable-js/issues{/number}",
			"pulls_url": "https://api.github.com/repos/facebook/immutable-js/pulls{/number}",
			"milestones_url": "https://api.github.com/repos/facebook/immutable-js/milestones{/number}",
			"notifications_url": "https://api.github.com/repos/facebook/immutable-js/notifications{?since,all,participating}",
			"labels_url": "https://api.github.com/repos/facebook/immutable-js/labels{/name}",
			"releases_url": "https://api.github.com/repos/facebook/immutable-js/releases{/id}",
			"deployments_url": "https://api.github.com/repos/facebook/immutable-js/deployments",
			"created_at": "2014-07-02T06:02:29Z",
			"updated_at": "2019-02-09T23:33:16Z",
			"pushed_at": "2019-01-30T10:34:36Z",
			"git_url": "git://github.com/facebook/immutable-js.git",
			"ssh_url": "git@github.com:facebook/immutable-js.git",
			"clone_url": "https://github.com/facebook/immutable-js.git",
			"svn_url": "https://github.com/facebook/immutable-js",
			"homepage": "http://facebook.github.io/immutable-js/",
			"size": 21746,
			"stargazers_count": 26421,
			"watchers_count": 26421,
			"language": "JavaScript",
			"has_issues": true,
			"has_projects": true,
			"has_downloads": true,
			"has_wiki": true,
			"has_pages": true,
			"forks_count": 1560,
			"mirror_url": null,
			"archived": false,
			"open_issues_count": 116,
			"license": {
				"key": "mit",
				"name": "MIT License",
				"spdx_id": "MIT",
				"url": "https://api.github.com/licenses/mit",
				"node_id": "MDc6TGljZW5zZTEz"
			},
			"forks": 1560,
			"open_issues": 116,
			"watchers": 26421,
			"default_branch": "master",
			"score": 1.0
		}
	]
}
