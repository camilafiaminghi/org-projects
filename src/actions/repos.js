export const REPOS_REQUEST = 'REPOS_REQUEST'
export const REPOS_SUCCESS = 'REPOS_SUCCESS'
export const REPOS_FAIL = 'REPOS_FAIL'

export function getRepos (org) {
	return {
		type: REPOS_REQUEST,
		payload: org
	}
}

export function getReposSuccess (repos) {
	return {
		type: REPOS_SUCCESS,
		payload: repos
	}
}

export function getReposFail (errors) {
	return {
		type: REPOS_FAIL,
		errors
	}
}
