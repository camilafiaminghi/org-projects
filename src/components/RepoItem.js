import React from 'react'
// import { NavLink } from 'react-router-dom'

export default function RepoItem ({ org, name, stargazers_count, description, language }) {
	return (
		<div>
			<a href="#">{name}</a>
			<p>{description}</p>
			<span>Language: {language}</span> | <span>Stars: {stargazers_count}</span>
		</div>
	)
}
