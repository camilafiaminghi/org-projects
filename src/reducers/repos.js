import { REPOS_REQUEST, REPOS_SUCCESS, REPOS_FAIL } from '../actions/repos'

export const initialState = {
	page: 1,
	sort: 'stars'
}

const repos = (state = {...initialState}, action) => {
	switch (action.type) {
		case REPOS_REQUEST:
			return {
				...state,
				...action.payload,
				items: [],
				loading: true
			}
		case REPOS_SUCCESS:
			const perPage = 4
			const nextPage = (action.payload.items % perPage === 0) ? false : true
			return {
				...state,
				...action.payload,
				nextPage,
				languages: [...new Set((action.payload.items || []).map(item => item.language))],
				errors: null,
				loading: false
			}
		case REPOS_FAIL:
			return {
				...state,
				errors: action.errors,
				loading: false
			}
		default:
			return state
	}
}

export default repos
