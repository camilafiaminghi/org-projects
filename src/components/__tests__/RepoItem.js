import React from 'react'
import { MemoryRouter } from 'react-router'
import { Link } from 'react-router-dom'
import renderer from 'react-test-renderer'
import RepoItem from './../RepoItem'
import repos from './../../__helpers__/repos'

const repo = repos.items[0]
let router
let wrapper

describe('<RepoItem />', () => {
	beforeEach(() => {
		const props = {
			...repo,
			org: repo.owner.login
		}
		router = shallow(<MemoryRouter><RepoItem {...props} /></MemoryRouter>)
		wrapper = router.find(RepoItem).shallow()
	})

	it('matches snapshot', () => {
		const props = {
			...repo,
			org: repo.owner.login
		}
		const component = renderer.create(<MemoryRouter><RepoItem {...props} /></MemoryRouter>)
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render', () => {
		expect(wrapper.exists()).toBeTruthy()
	})

	it('should has a link', () => {
		expect(wrapper.find('Link').exists()).toBeTruthy()
	})

	it('should has a description', () => {
		expect(wrapper.find('p').exists()).toBeTruthy()
		expect(wrapper.find('p').shallow().props().children).toEqual(repo.description)
	})

	it('should has languare and starred count', () => {
		expect(wrapper.find('span').exists()).toBeTruthy()
		expect(wrapper.find('span').get(0).props.children[1]).toEqual(<strong>{repo.language}</strong>)
		expect(wrapper.find('span').get(1).props.children[1]).toEqual(<strong>{repo.stargazers_count}</strong>)
	})
})
