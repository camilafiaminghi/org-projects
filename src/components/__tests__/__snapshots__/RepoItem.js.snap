// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoItem /> matches snapshot 1`] = `
<div>
  <a
    aria-label="Go to react branches"
    href="/branches/facebook/react"
    onClick={[Function]}
  >
    react
  </a>
  <p>
    A declarative, efficient, and flexible JavaScript library for building user interfaces.
  </p>
  <span>
    Language: 
    JavaScript
  </span>
   | 
  <span>
    Stars: 
    122084
  </span>
</div>
`;
